ENH: organize - show files with conflicts requiring adding _obj-

yarikoptic opened the pull request:
Logging check is currently not really functioning since IIRC we do enable logging into a file at higher level than INFO. So we might need to tune decision making here

Should provide UX side to partially address:
- https://github.com/dandi/dandi-cli/issues/1489

example:

```
❯ dandi organize --dandiset-path /tmp/outds /home/yoh/proj/dandi/dandisets/000004/sub-P53CS/*nwb /home/yoh/proj/dandi/dandisets/000004/sub-P54CS/*nwb
...
2024-08-19 20:03:17,448 [    INFO] Symlink support autodetected; setting files_mode='symlink'
2024-08-19 20:03:17,452 [    INFO] 2 out of 5 paths are not unique.
 2 paths 'compete' for the path 'sub-P53CS/sub-P53CS_ses-20171101_ecephys+image.nwb':
  /home/yoh/proj/dandi/dandisets/000004/sub-P53CS/sub-P53CS_ses-20171101_obj-108aqix_ecephys+image.nwb
  /home/yoh/proj/dandi/dandisets/000004/sub-P53CS/sub-P53CS_ses-20171101_obj-lj04dr_ecephys+image.nwb
 3 paths 'compete' for the path 'sub-P54CS/sub-P54CS_ses-20180101_ecephys+image.nwb':
  /home/yoh/proj/dandi/dandisets/000004/sub-P54CS/sub-P54CS_ses-20180101_obj-1ru98u8_ecephys+image.nwb
  /home/yoh/proj/dandi/dandisets/000004/sub-P54CS/sub-P54CS_ses-20180101_obj-1ukzk61_ecephys+image.nwb
  /home/yoh/proj/dandi/dandisets/000004/sub-P54CS/sub-P54CS_ses-20180101_obj-1vx1a5w_ecephys+image.nwb
 We will try adding _obj- based on crc32 of object_id
```

TODOs
- [ ] check what leads to tests failing ATM
- [ ] extend test covering this code to have multiple groups requiring `_obj`

yarikoptic committed:
ENH: organize - show files with conflicts requiring adding _obj-

Logging check is currently not really functioning since IIRC we
do enable logging into a file at higher level than INFO. So we
might need to tune decision making here

yarikoptic commented on dandi/organize.py:
```diff
@@ -307,9 +308,28 @@ def _assign_obj_id(metadata, non_unique):
+
+    msg += "\n " + get_msg(non_unique_paths[0])
+    if len(non_unique) > 1:
+        if not lgr.isEnabledFor(logging.DEBUG):
```
@jwodder 
this condition is not effective -- we seems always enabled and effectiveLevel is 0.  I guess due to having multiple backends etc. Is there a way to make it dependent on what user specifies in cli?

jwodder replied:
I don't think this condition is a good idea in the first place; the content of log messages should not depend on whether another log messages would be shown.  If you really want to display some paths at INFO and the full paths at DEBUG, then always log two messages.

jwodder commented on dandi/organize.py:
```diff
@@ -307,9 +308,28 @@ def _assign_obj_id(metadata, non_unique):
     from .pynwb_utils import get_object_id
 
-    msg = "%d out of %d paths are not unique" % (len(non_unique), len(metadata))
+    msg = "%d out of %d paths are not unique." % (len(non_unique), len(metadata))
```
Use f-strings here.

yarikoptic replied:
yeah, should be consistent.  
```suggestion
    msg = f"{len(non_unique)} out of {len(metadata)} paths are not unique."
```

jwodder commented on dandi/organize.py:
```diff
@@ -307,9 +308,28 @@ def _assign_obj_id(metadata, non_unique):
 
-    lgr.info(msg + ". We will try adding _obj- based on crc32 of object_id")
+    # provide more information to the user
+    def get_msg(path, indent="  "):
```
`get_msg()` never seems to be called with a custom `indent`, so why make it a parameter?

yarikoptic replied:
yeah, in initial version I had different indents, will change

jwodder commented on dandi/organize.py:
```diff
@@ -307,9 +308,28 @@ def _assign_obj_id(metadata, non_unique):
+        else:
+            for ex_path in non_unique_paths[1:]:
+                msg += "\n " + get_msg(ex_path)
+    lgr.info(msg + "\n We will try adding _obj- based on crc32 of object_id")
```
```suggestion
    lgr.info("%s\n We will try adding _obj- based on crc32 of object_id", msg)
```

This will prevent crashes if a nonunique path contains `%`.

yarikoptic replied:
I think we figured that without `args` provided, it would not try to do interpolation, but I agree that this would be a safer pattern, I will apply.

yarikoptic committed:
Small tune ups from code review

Co-authored-by: John T. Wodder II <jwodder@users.noreply.github.com>

yarikoptic committed:
RF: make logging messages independent of active level - 1: hits always at DEBUG

With this change logging looks like

    2024-08-21 09:42:39,718 [    INFO] 2 out of 5 paths are not unique.
     2 paths 'compete' for the path 'sub-P53CS/sub-P53CS_ses-20171101_ecephys+image.nwb':
      /home/yoh/proj/dandi/dandisets/000004/sub-P53CS/sub-P53CS_ses-20171101_obj-108aqix_ecephys+image.nwb
      /home/yoh/proj/dandi/dandisets/000004/sub-P53CS/sub-P53CS_ses-20171101_obj-lj04dr_ecephys+image.nwb
     1 more case(s) are listed at DEBUG level.
    2024-08-21 09:42:39,718 [   DEBUG] Additional non-unique paths:
     3 paths 'compete' for the path 'sub-P54CS/sub-P54CS_ses-20180101_ecephys+image.nwb':
      /home/yoh/proj/dandi/dandisets/000004/sub-P54CS/sub-P54CS_ses-20180101_obj-1ru98u8_ecephys+image.nwb
      /home/yoh/proj/dandi/dandisets/000004/sub-P54CS/sub-P54CS_ses-20180101_obj-1ukzk61_ecephys+image.nwb
      /home/yoh/proj/dandi/dandisets/000004/sub-P54CS/sub-P54CS_ses-20180101_obj-1vx1a5w_ecephys+image.nwb
    2024-08-21 09:42:39,718 [    INFO] We will try adding _obj- based on crc32 of object_id
